### PR TITLE
Refactor the KubeVirt dns settings

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Refactor Kubevirt DNSConfig type from being a raw string to PodDNSConfig. This config is part of kubermatic datacenter configs.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Follow up PR to this issue https://github.com/kubermatic/kubermatic/issues/6121

**Optional Release Note**:
```release-note
Support Kubevirt DNSConfig in Kubermatic 
```

### CODE REVIEW

**Review by kron4eg — APPROVED** [review]

**Comment by kron4eg** [comment]
/hold

until https://github.com/kubermatic/machine-controller/pull/847 is fixed

**Comment by moadqassem** [comment]
@kron4eg Can I get another LGTM?

**Review by kron4eg — APPROVED** [review]

**Pull request merged**
